### PR TITLE
Added Counting Sort [JavaScript]

### LINKED ISSUE #378: Counting Sort [JavaScript]
#### This issue is about [Counting Sort](http://www.geeksforgeeks.org/counting-sort/).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by Anshuman-Verma**
Fixes #378 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by Anshuman-Verma** [comment]
**DO NOT REVIEW**

File: counting_sort/countingSort.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+*  max is the maximum element in array
+*/
+
+function countingSort (arr, max) {
+	/*
+	:param arr: Array to be sorted
+	:param max: Maximum value in array
+	:return: Sorted array
+	*/
+	let i = 0;
```
Comment (mohitkyadav) [inline counting_sort/countingSort.js]: The scope of `i` can be reduced.

**Comment by Anshuman-Verma** [comment]
Fixed @mohitkyadav

File: counting_sort/countingSort.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/* Following algorithm sorts the input array in ascending order
+*  Time complexity is O(n+k)
+*  Auxiliary space is O(n+k)
```
Comment (mohitkyadav) [inline counting_sort/countingSort.js]: Please add information about `n` and `k`.

Comment (Anshuman-Verma) [inline counting_sort/countingSort.js]: Look at the 4th line

Comment (mohitkyadav) [inline counting_sort/countingSort.js]: I didn't see that. :laughing:

Comment (Anshuman-Verma) [inline counting_sort/countingSort.js]: Haha, no worries :stuck_out_tongue:

File: counting_sort/countingSort.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+*/
+
+function countingSort (arr, max) {
+	/*
+	:param arr: Array to be sorted
+	:param max: Maximum value in array
+	:return: Sorted array
+	*/
+	let n = arr.length;
+	console.log(max);
```
Comment (aviaryan) [inline counting_sort/countingSort.js]: A debug statement? Not needed.

Comment (Anshuman-Verma) [inline counting_sort/countingSort.js]: Oh :disappointed:  , now it's taken care of.

**Pull request merged**